### DISCUSSION TITLE
Block third-party connections to a TLD with an exception

### DISCUSSION THREAD

**Discussion by ghost** [uBlockOrigin/uBlock-issues#2412] [Q&A] [answered]
Can this be rewritten without whitelisting `example.it` altogether?

    ||it$3p
    @@||example.it$3p

**Comment by MasterKia** [comment]
`||it$3p,domain=~example.it`

https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#domain

**Reply by ghost** [reply]
I want to unblock third-party connections to `example.it`, not from `example.it`. The actual domain is a CDN service.

**Reply by ghost** [reply]
/https?:\/\/[^/]*(?<!\bexample)\.it\//$3p

?

***

Will match `bad-example.it`.

**Comment by gorhill** [comment, marked as answer]
To be on par with DNR API, I have been considering supporting `requestdomain=` (or a better suggested name), and in this case the filter would be:

    *$3p,requestdomain=it|~example.it

I will decide the next dev cycle about this.

**Reply by gorhill** [reply]
Actually a better filter performance-wise would be:

    ||it^$3p,requestdomain=~example.it

Since the keyword `it` can be extracted from it.

**Reply by uBlock-user** [reply]
@gorhill When I add `||google.com^$to=com` or `||google.com^$from=com` navigation is blocked by document-blocked page. Is this a bug too like the one with `$method` ?

**Reply by gorhill** [reply]
I get the same result with `||google.com^$domain=com` in 1.45.2, so not something new. That sort of filter is not very realistic though, not sure I want to throw code at this to avoid document-blocked page for such unrealistic filters.

**Reply by uBlock-user** [reply]
Yeah I came across this when I was testing `$to` and `$from` with all kinds of filters, I never deploy such filters myself though.

**Comment by MasterKia** [comment]
https://github.com/uBlockOrigin/uBlock-issues/wiki/Static-filter-syntax#from
https://github.com/uBlockOrigin/uBlock-issues/wiki/Static-filter-syntax#domain

https://github.com/uBlockOrigin/uBlock-issues/wiki/Static-filter-syntax#to
https://github.com/uBlockOrigin/uBlock-issues/wiki/Static-filter-syntax#denyallow
